src/vs/workbench/contrib/terminal/browser/terminalInstance.ts

```diff
@@ -541,10 +543,30 @@ export class TerminalInstance extends Disposable implements ITerminalInstance {
 				event.preventDefault();
 				return false;
 			}
-
-			// Skip processing by xterm.js of keyboard events that resolve to commands described
-			// within commandsToSkipShell
-			if (resolveResult && this._skipTerminalCommands.some(k => k === resolveResult.commandId)) {
+			const promptChoices: IPromptChoice[] = [
+				{
+					label: nls.localize('configureTerminalSettings', "Configure Terminal Settings"),
+					run: () => {
+						this._commandService.executeCommand('workbench.action.openSettings', '@id:terminal.integrated.commandsToSkipShell,terminal.integrated.sendToShell,terminal.integrated.allowChords');
```

Comment:
I think there is a method on PreferencesService that will let you do this, you shouldn't have to call executeCommand most of the time